core/client/src/test/java/alluxio/client/file/policy/DeterministicHashPolicyTest.java

```diff
@@ -0,0 +1,106 @@
+        .setRpcPort(PORT).setDataPort(PORT).setWebPort(PORT), 2 * (long) Constants.GB, 0));
+    mWorkerInfos.add(new BlockWorkerInfo(new WorkerNetAddress().setHost("worker3")
+        .setRpcPort(PORT).setDataPort(PORT).setWebPort(PORT), 3 * (long) Constants.GB, 0));
+    mWorkerInfos.add(new BlockWorkerInfo(new WorkerNetAddress().setHost("worker4")
+        .setRpcPort(PORT).setDataPort(PORT).setWebPort(PORT), 3 * (long) Constants.GB, 0));
+  }
+
+  @Test
+  public void getWorkerDeterministically() {
+    DeterministicHashPolicy policy = (DeterministicHashPolicy) BlockLocationPolicy.Factory.create(
+        BlockLocationPolicyCreateOptions.defaults()
+            .setLocationPolicyClassName("alluxio.client.file.policy.DeterministicHashPolicy"));
+    String host = policy.getWorkerForBlock(
+        BlockLocationPolicyGetWorkerOptions.defaults().setBlockWorkerInfos(mWorkerInfos)
+            .setBlockId(1).setBlockSize(2 * (long) Constants.GB)).getHost();
+    for (int i = 0; i < 10; i++) {
+      DeterministicHashPolicy p = (DeterministicHashPolicy) BlockLocationPolicy.Factory.create(
+          BlockLocationPolicyCreateOptions.defaults()
+              .setLocationPolicyClassName("alluxio.client.file.policy.DeterministicHashPolicy"));
+      // For the same block, always return the same worker.
+      Assert.assertEquals(host, p.getWorkerForBlock(
+          BlockLocationPolicyGetWorkerOptions.defaults().setBlockWorkerInfos(mWorkerInfos)
+              .setBlockId(1).setBlockSize(2 * (long) Constants.GB)).getHost());
+      Assert.assertEquals(host, p.getWorkerForBlock(
+          BlockLocationPolicyGetWorkerOptions.defaults().setBlockWorkerInfos(mWorkerInfos)
+              .setBlockId(1).setBlockSize(2 * (long) Constants.GB)).getHost());
+    }
+  }
+
+  @Test
+  public void getWorkerEnoughCapacity() {
+    DeterministicHashPolicy policy = (DeterministicHashPolicy) BlockLocationPolicy.Factory.create(
+        BlockLocationPolicyCreateOptions.defaults()
+            .setLocationPolicyClassName("alluxio.client.file.policy.DeterministicHashPolicy"));
+    for (long blockId = 0; blockId < 100; blockId++) {
+      // worker1 does not have enough capacity. It should never be picked.
+      Assert.assertNotEquals("worker1", policy.getWorkerForBlock(
+          BlockLocationPolicyGetWorkerOptions.defaults().setBlockWorkerInfos(mWorkerInfos)
+              .setBlockId(blockId).setBlockSize(2 * (long) Constants.GB)).getHost());
+    }
+  }
+
+  @Test
+  public void getWorkerMultipleShards() {
+    DeterministicHashPolicy policy2 = (DeterministicHashPolicy) BlockLocationPolicy.Factory.create(
+        BlockLocationPolicyCreateOptions.defaults()
+            .setLocationPolicyClassName("alluxio.client.file.policy.DeterministicHashPolicy")
+            .setDeterministicHashPolicyNumShards(2));
+    Set<String> addresses1 = new HashSet<>();
+    Set<String> addresses2 = new HashSet<>();
+    for (int i = 0; i < 100; i++) {
```

Comment:
The policy randomly picks 1 host among 2. This 100 loops makes sure that both of the 2 hosts are picked overtime.